### PR TITLE
Nnbd migration.

### PR DESCRIPTION
**Pull request by bsutton**
I would be great if you can take this PR which migrates google_signin_mock to null safety.

Thanks,
Brett

### CODE REVIEW

File: lib/src/google_sign_in_mocks_base.dart [hunk 1]
```diff
@@ -18,6 +18,13 @@ class MockGoogleSignInAccount extends Mock implements GoogleSignInAccount {
   @override
   Future<GoogleSignInAuthentication> get authentication =>
       Future.value(MockGoogleSignInAuthentication());
+
+  @override
+  bool operator ==(other) {
```
Comment (atn832) [inline lib/src/google_sign_in_mocks_base.dart:23]: Do we need this? It seems like this is the exact same code as Mock's == code: https://github.com/dart-lang/mockito/blob/171d4d60f59826c0c1c84ddb98b6161474a80bea/lib/src/mock.dart#L195

I'd rather we didn't add in this PR it if it is not needed for non-nullability.

**Comment by atn832** [comment]
Looks good, thanks @bsutton! I just have one question about `==`

**Comment by bsutton** [comment]
The mock operator failed as it declares the argument as 'object' whilst the Google class declares it with dynamic.

So this was my work around. I'm guessing the google library needs updating.

**Comment by atn832** [comment]
Oh ok! You're right, we have no choice but to follow Google Sign In's interface.

I wonder why Google Sign In went for dynamic. It looks like Object is the preferred choice over dynamic:
- https://dart.dev/guides/language/effective-dart/design#dont-make-the-parameter-to--nullable
- https://dart.dev/null-safety/understanding-null-safety#never-for-unreachable-code
- https://github.com/dart-lang/sdk/issues/42687#issuecomment-658119834

**Comment by atn832** [comment]
I published your PR to https://pub.dev/packages/google_sign_in_mocks/changelog#020. Apparently Mockito doesn't support full null safety yet.

**Pull request merged**